selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/dashboard/workspaces/details/WorkspaceDetailsMachineActionsTest.java

```diff
@@ -0,0 +1,267 @@
+/*
+ * Copyright (c) 2012-2018 Red Hat, Inc.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.selenium.dashboard.workspaces.details;
+
+import static org.eclipse.che.selenium.pageobject.dashboard.workspaces.WorkspaceDetails.ActionButton.APPLY_BUTTON;
+import static org.eclipse.che.selenium.pageobject.dashboard.workspaces.WorkspaceDetails.ActionButton.CANCEL_BUTTON;
+import static org.eclipse.che.selenium.pageobject.dashboard.workspaces.WorkspaceDetails.ActionButton.SAVE_BUTTON;
+import static org.eclipse.che.selenium.pageobject.dashboard.workspaces.WorkspaceDetails.WorkspaceDetailsTab.MACHINES;
+import static org.openqa.selenium.Keys.ARROW_DOWN;
+import static org.openqa.selenium.Keys.ARROW_UP;
+import static org.openqa.selenium.Keys.ESCAPE;
+import static org.testng.Assert.fail;
+
+import com.google.inject.Inject;
+import org.eclipse.che.commons.lang.NameGenerator;
+import org.eclipse.che.selenium.core.SeleniumWebDriver;
+import org.eclipse.che.selenium.core.webdriver.SeleniumWebDriverHelper;
+import org.eclipse.che.selenium.core.workspace.TestWorkspace;
+import org.eclipse.che.selenium.pageobject.dashboard.Dashboard;
+import org.eclipse.che.selenium.pageobject.dashboard.workspaces.EditMachineForm;
+import org.eclipse.che.selenium.pageobject.dashboard.workspaces.WorkspaceDetails;
+import org.eclipse.che.selenium.pageobject.dashboard.workspaces.WorkspaceDetailsMachines;
+import org.eclipse.che.selenium.pageobject.dashboard.workspaces.WorkspaceEnvVariables;
+import org.eclipse.che.selenium.pageobject.dashboard.workspaces.WorkspaceInstallers;
+import org.eclipse.che.selenium.pageobject.dashboard.workspaces.WorkspaceServers;
+import org.eclipse.che.selenium.pageobject.dashboard.workspaces.Workspaces;
+import org.openqa.selenium.TimeoutException;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+public class WorkspaceDetailsMachineActionsTest {
+  private static final String MACHINE_NAME = "dev-machine";
+  private static final String CHANGED_MACHINE_NAME = "test-machine";
+  private static final String IMAGE_NAME = "eclipse/ubuntu_jdk8";
+  private static final String EXPECTED_RAM_VALUE = "2";
+  private static final String EMPTY_NAME_ERROR_MESSAGE = "Machine's name is required.";
+  private static final String SPECIAL_CHARACTERS_ERRORS_MESSAGE =
+      "The name should not contain special characters like space, dollar, etc.";
+  private static final String NAME_WITH_SPECIAL_CHARACTERS = "@#$^&*(!";
+  private static final String MAX_VALID_NAME = NameGenerator.generate("max_name", 120);
+  private static final String INVALID_LENGTH_NAME = NameGenerator.generate(MAX_VALID_NAME, 1);
```

Comment:
too small length is invalid as well.  So more straightforward name explain the value of constant clearer: `TOO_BIG_NAME`